kylin-spark-project/kylin-spark-common/src/main/scala/org/apache/spark/sql/execution/datasource/FilePruner.scala

```diff
@@ -170,7 +172,8 @@ class FilePruner(cubeInstance: CubeInstance,
   }
 
   private def genShardSpec(selected: Seq[SegmentDirectory]): Option[ShardSpec] = {
-    if (!KylinConfig.getInstanceFromEnv.isShardingJoinOptEnabled || selected.isEmpty) {
+    val isJoinQuery = OLAPContext.getThreadLocalContexts.map(_.allOlapJoins).exists(_.nonEmpty);
```

Comment:
This way is wrong, for example:
```
select XXX from fact_table left join join_table on ... 
```
In this case shown above, the value of 'isJoinQuery' is true, but the join operator will be removed.